### PR TITLE
[build-script] Disable pkg-config for Darwin hosts (NFC)

### PR DESCRIPTION
**Pull request by kastiglione**
Disable `pkg-config` on Darwin to prevent compilation issues.

This is an NFC follow up to https://github.com/apple/swift/pull/32436 and https://github.com/apple/swift/pull/32437.

The benefits are:

1. Prevents the `pkg-config` issues directly, instead of a roundabout way
2. Centralize two related pieces of build isolation config to a single place

For 1, the original issue was that CMake `find_` functions would search `pkg-config` before the SDK. Homebrew `pkg-config` hardcodes CommandLineTools, even when Xcode is installed, and this caused problems. See [SR-12726](https://bugs.swift.org/browse/SR-12726).

Disabling `pkg-config` gives better isolation, and it removes the use of `CMAKE_PREFIX_PATH` which I see as a bigger hammer than disabling `pkg-config`.

Before any of these changes, if `pkg-config` is installed, the effective search order was:

1. Try `CMAKE_PREFIX_PATH` (empty)
2. Try `HINTS` paths (CommandLineTools via `pkg-config`)
3. Try SDK paths

After #32436, the relevant search order became:

1. Try `CMAKE_PREFIX_PATH` (SDK)
2. Try `HINTS` paths (CommandLineTools via `pkg-config`)
3. Try SDK paths

Now with this change, the relevant search order is:

1. Try `CMAKE_PREFIX_PATH` (empty)
2. Try `HINTS` paths (empty)
3. Try SDK paths

### CODE REVIEW

File: utils/build-script [hunk 1]
```diff
@@ -655,9 +655,12 @@ class BuildScriptInvocation(object):
             ]
 
             # Isolate build from the system; Darwin toolchains build against SDKs.
-            args.extra_cmake_options.append(
-                '-DCMAKE_IGNORE_PATH=/usr/lib;/usr/local/lib;/lib'
-            )
+            # For additional isolation, disable pkg-config. Homebrew's pkg-config
+            # prioritizes CommandLineTools paths, resulting in compile errors.
+            args.extra_cmake_options += [
+                '-DCMAKE_IGNORE_PATH=/usr/lib;/usr/local/lib;/lib',
```
Comment (gottesmm) [inline utils/build-script:661]: Is there a cmake option we can use that just does this without us needing to hack around it?

@compnerd

Comment (kastiglione) [reply utils/build-script:661]: I spoke with @gottesmm, he's referring to the line below:

```python
'-DPKG_CONFIG_EXECUTABLE=/usr/bin/true',
```

Comment (kastiglione) [reply utils/build-script:661]: This was discussed out of band, and the conclusion was that disabling `pkg-config` via `PKG_CONFIG_EXECUTABLE=/usr/bin/false` is the most direct & concise way to do this. A larger scoped solution is to use checked in CMake cache files to explicitly declare SDK dependencies, which may happen in the future.

**Review by gottesmm — APPROVED** [review]
LGTM

**Comment by kastiglione** [comment]
@swift-ci please smoke test and merge

**Pull request merged**